devtools/maven/src/main/java/io/quarkus/maven/GenerateCodeTestsMojo.java

```diff
@@ -1,16 +1,107 @@
 package io.quarkus.maven;
 
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
 import org.apache.maven.plugins.annotations.ResolutionScope;
 
+import io.quarkus.bootstrap.app.CuratedApplication;
+import io.quarkus.builder.Json;
+import io.quarkus.maven.dependency.ResolvedDependency;
+import io.quarkus.runtime.LaunchMode;
+
 @Mojo(name = "generate-code-tests", defaultPhase = LifecyclePhase.GENERATE_TEST_SOURCES, requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME, threadSafe = true)
 public class GenerateCodeTestsMojo extends GenerateCodeMojo {
     @Override
     protected void doExecute() throws MojoExecutionException, MojoFailureException {
         generateCode(getParentDirs(mavenProject().getTestCompileSourceRoots()),
                 path -> mavenProject().addTestCompileSourceRoot(path.toString()), true);
+
+        final String testProfile = System.getProperty("quarkus.test.integration-test-profile");
+        if ("test-with-native-agent".equals(testProfile)) {
+            generateNativeAgentFilters();
+        }
+    }
+
+    private void generateNativeAgentFilters() throws MojoExecutionException {
+        getLog().debug("Generate native image agent filters");
+
+        // Get packages to exclude
+        Collection<String> commonExcludePackageNames = getPackageNames();
+
+        // Generate json using the packages
+        generateNativeAgentFilter(getCallerExcludePackageNames(commonExcludePackageNames),
+                Path.of(mavenProject().getModel().getBuild().getDirectory(),
+                        "quarkus-caller-filter.json"));
+        generateNativeAgentFilter(getAccessExcludePackageNames(commonExcludePackageNames),
+                Path.of(mavenProject().getModel().getBuild().getDirectory(),
+                        "quarkus-access-filter.json"));
+    }
+
+    private Collection<String> getAccessExcludePackageNames(Collection<String> commonExcludePackageNames) {
+        final Set<String> result = new HashSet<>(commonExcludePackageNames);
+        result.add("javax.crac");
+        result.add("jdk.crac");
+        return result;
+    }
+
+    private Set<String> getCallerExcludePackageNames(Collection<String> commonExcludePackageNames) {
+        final Set<String> result = new HashSet<>(commonExcludePackageNames);
+        result.add("com.sun");
+        result.add("java");
+        result.add("javax");
+        result.add("jdk");
+        result.add("sun");
```

Comment:
I selected these specific packages based on the getting started Quarkus application, so one of the most basic Quarkus applications. Without these exclusions some JNI and reflection configuration was being added which seemed unnecessary. See the following attached files for examples:

[jni-origins.txt](https://github.com/quarkusio/quarkus/files/14943352/jni-origins.txt)
[reflect-origins.txt](https://github.com/quarkusio/quarkus/files/14943356/reflect-origins.txt)

These specific exclusions might be too restrictive for some other use cases, e.g. AWT/images, where reflection or library load calls come from inside the JDK. I didn't want to make these hardcoded exclusions very fine grained, so I think we can go 2 ways here:

a) Keep these wide exclusions and revisit them as needed as we try different use cases.
b) Remove these wide exclusions and accept some extra unnecessary configuration.

It might be safer to go with b) here.

Reply:
I think we should do some tests to see what's the actual impact of b) on the image size, and if possible on runtime performance.

Since the agent detects that these are actually accessed I am curious how/why we can get away without them.

Reply:
> Sidenote: Since the agent detects that these are actually accessed I am curious how/why we can get away without them.

In some cases svm can do constant folding of those, e.g.
```
               └── io.quarkus.arc.impl.ArcContainerImpl#<init>(io.quarkus.arc.CurrentContextFactory,boolean)
                   └── java.util.concurrent.atomic.AtomicBoolean#<clinit>()
                       └── java.lang.invoke.MethodHandles$Lookup#findVarHandle(java.lang.Class,java.lang.String,java.lang.Class) - [ {   "name":"java.util.concurrent.atomic.AtomicBoolean",   "fields":[{"name":"value"}] } ]
s":[{"name":"value"}] } ]
```

SVM handles use cases like this by transforming the graph by doing

Reply:
Also, once you have something running and you understand the configuration, you can see if you want to trim back, say to speed up build times or reduce image size.

Reply:
I've removed JDK package filter exclusions and added some comments. The extra configuration added for the getting started Quarkus app is quite small so it's a reasonable thing to do.